rules/rbe_repo.bzl

```diff
@@ -943,6 +958,28 @@ def rbe_autoconfig(
         use_checked_in_confs = use_checked_in_confs,
     )
 
+    # If create_java_configs was requested but no java_home or detect_java_home was
+    # set, we try to resolve a java_home
+    if create_java_configs and not java_home and not detect_java_home:
+        # If a spec was found and that has a java_home, use it
+        if toolchain_config_spec and toolchain_config_spec.create_cc_configs:
+            java_home = toolchain_config_spec.java_home
+
+        elif toolchain_config_suite_spec.get("default_java_home"):
+            # Fallback to try to using the default_java_home set in the
+            # toolchain_config_suite_spec
+            java_home = toolchain_config_suite_spec.get("default_java_home")
+
+        elif use_checked_in_confs != CHECKED_IN_CONFS_FORCE:
+            # Fallback to detecting the java_home if CHECKED_IN_CONFS_FORCE
+            # was not passed
+            detect_java_home = True
+
+        elif toolchain_config_spec and use_checked_in_confs == CHECKED_IN_CONFS_FORCE:
```

Comment:
Not fully understand this part.

Reply:
changed to a fail early and documented better, thanks for noting